Auto ivc2 mods

Kenneth-T-Moore opened the pull request:
1. Started working on api translation diagram.
2. Converted simple tutorials to new ivc-less config.
3. Modified Driver to always store auto-ivc desvars by their promoted input instead of ivc output.
4. Modified case recording to always store auto-ivc desvars by their promoted input instead of ivc output.

DKilkenny committed:
Added om.slicer support to indices

DKilkenny committed:
Added om.slicer support to indices in add_response and add_constraint

DKilkenny committed:
pep

DKilkenny committed:
Cleaned up test

DKilkenny committed:
Bret comments 1/2

Commit:
Merge branch 'auto_ivc2' of github.com:naylor-b/openmdao into auto_ivc2

Commit:
cleanup

DKilkenny committed:
Bret's comments part 2/2

DKilkenny committed:
Bret's comments part 2/2

DKilkenny committed:
added resid checks to docs and couple lines on why in implicit comp doc

johnjasa committed:
Added iter_count_without_approx

johnjasa committed:
Merge branch 'master' of https://github.com/OpenMDAO/OpenMDAO into approx_flag

Commit:
all tests pass at present (except maybe gui)

johnjasa committed:
Modified test for iter_count_without_approx

johnjasa committed:
Added docstring for under_approx

Commit:
Merge branch 'auto_ivc2' of github.com:naylor-b/openmdao into auto_ivc2

DKilkenny committed:
removed temp variables

Commit:
Checkpoint.

DKilkenny committed:
test fixes

swryan committed:
multi mvp, test

johnjasa committed:
Fixing pep8 lint

hschilling committed:
pass environment variables to subprocess call to avoid errors when running mpi commands during doc builds

naylor-b commented on openmdao/core/system.py:
```diff
@@ -2877,11 +2900,31 @@ def get_responses(self, recurse=True, get_sizes=True):
 
         # Human readable error message during Driver setup.
         try:
             out = OrderedDict((prom2abs[name][0], data) for name, data in
```
Since you've added the loop below, you could probably just do `out = OrderedDict()` here.  Otherwise I think you're needlessly looping over the responses twice.

Kenneth-T-Moore replied:
I took that line out a little bit ago. It was a bug, and not meant to be there.

DKilkenny committed:
Bret's comments

naylor-b commented on openmdao/recorders/sqlite_recorder.py:
```diff
@@ -344,6 +347,11 @@ def startup(self, recording_requester):
             for var_set, var_type in full_var_set:
                 for name in var_set:
+
+                    # Design variables are always requested by input name.
```
always?

Kenneth-T-Moore replied:
I guess not always, since you can still do the manual ivc.

swryan committed:
Merge pull request #1490 from tadkollar/n2compress1489

Model data now compressed when saved in N2 HTML file

DKilkenny committed:
External code comp docs

Commit:
Next batch of tests pass

DKilkenny committed:
ExternalCodeImplicitComp

DKilkenny committed:
LinearSystemComp

swryan committed:
mvp checks/tests

DKilkenny committed:
MetamodelStructuredComp

DKilkenny committed:
MetamodelStructuredComp

DKilkenny committed:
MetamodelUnStructuredComp

Commit:
Test for Bret

Commit:
fixed a couple tests

swryan committed:
dpc w tests

DKilkenny committed:
Demux/Mux Comp

Commit:
a couple tests

swryan committed:
lint

naylor-b committed:
fixed distrib dv bug

Commit:
another few tests

swryan committed:
docs in preogress

swryan committed:
enable CI on auto_ivc branch

naylor-b committed:
removed conflicts

swryan committed:
enable appveyor CI on auto_ivc branch

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc2

Kenneth-T-Moore committed:
Merge pull request #6 from naylor-b/ken_auto_ivc

fixed distrib dv bug

Commit:
Merge branch 'auto_ivc2' of github.com:Kenneth-T-Moore/OpenMDAO into auto_ivc2

DKilkenny committed:
Merge branch 'auto_ivc2' of github.com:/Kenneth-T-Moore/blue into auto_ivc_docs

DKilkenny committed:
KSComp

swryan committed:
Merge pull request #1494 from naylor-b/ken_auto_ivc

Added automatic creation of IndepVarComps for all unconnected inputs.

naylor-b committed:
appveyor test fixes

swryan committed:
rename args

DKilkenny committed:
Added slicer func and 3d array test

swryan committed:
Merge pull request #1495 from naylor-b/ken_auto_ivc

appveyor test fixes

DKilkenny committed:
used _var_allprocs_abs2meta for MPI cases

DKilkenny committed:
merged in auto_ivc branch to this issue

Commit:
another problem with a parallel case

DKilkenny committed:
using global size and shape

naylor-b committed:
fixed auto_ivc remote connection issue

Kenneth-T-Moore committed:
Merge pull request #8 from naylor-b/ken_auto_ivc

fixed auto_ivc remote connection issue

DKilkenny committed:
pep

Kenneth-T-Moore committed:
Merge pull request #7 from DKilkenny/auto_ivc_docs

Auto ivc component docs

swryan committed:
fix doc

swryan committed:
Merge pull request #1496 from swryan/docfix

fix doc

DKilkenny committed:
working

DKilkenny committed:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into guess_nonlinear_docs

tadkollar committed:
Added special coloring for inputs with an auto-IVC'd source

swryan committed:
Merge pull request #1460 from DKilkenny/guess_nonlinear_docs

Adding apply_nonlinear to _guess_nonlinear

tadkollar committed:
Changed auto-icv'd color to red

tadkollar committed:
Turned off debug info

DKilkenny committed:
Added distributed test

DKilkenny committed:
Merge of Auto_IVC

hschilling committed:
merged out

hschilling committed:
Merge branch 'hschilling-i1482-system-option-recording' into auto_ivc

naylor-b committed:
fix for slice conversion

DKilkenny committed:
Merge pull request #12 from naylor-b/danny_slice

fix for slice conversion

DKilkenny committed:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into parallel_indices_test

DKilkenny committed:
working

DKilkenny committed:
temp variables and negative stop tests

Commit:
fixed a few more tests

DKilkenny committed:
SplineComp

DKilkenny committed:
AddSubtractComp

swryan committed:
Merge pull request #1491 from DKilkenny/slice

Added om.slicer support for all src_indices/indices arguments

DKilkenny committed:
CrossProductComp

DKilkenny committed:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into parallel_indices_test

DKilkenny committed:
Merge out with failing test

swryan committed:
sync args/opts

DKilkenny committed:
DotProductComp

Commit:
todo

DKilkenny committed:
MatrixVectorProductComp

DKilkenny committed:
VectorMagnitudeComp

DKilkenny committed:
removed duplicate Spline comp in docs

DKilkenny committed:
Merge branch 'auto_ivc2' of github.com:/Kenneth-T-Moore/blue into auto_ivc_docs

tadkollar committed:
Don't include _auto_ivc as a tree component

tadkollar committed:
Merge pull request #93 from OpenMDAO/auto_ivc

Auto ivc

tadkollar committed:
Merge pull request #94 from tadkollar/master

sync

tadkollar committed:
Pulled latest

naylor-b committed:
updated multi_proc_exception_check

DKilkenny committed:
Merge pull request #13 from naylor-b/danny_err_msg

updated multi_proc_exception_check

Commit:
Par derivs roadblock.

swryan committed:
update cross_product & vector_magnitude comps

swryan committed:
Merge pull request #1484 from DKilkenny/parallel_indices_test

Added test from Ben Brelje

DKilkenny committed:
Added testcase

DKilkenny committed:
Merge branch 'auto_ivc2' of github.com:/Kenneth-T-Moore/blue into auto_ivc_docs

DKilkenny committed:
nonlinear_block_gs

swryan committed:
Merge branch 'master' into work

DKilkenny committed:
NonlinearBlockJac and RunOnce

Commit:
Auto IVC bug fixes - set 1

Commit:
Auto IVC bug fixes - set 1

DKilkenny committed:
Newton

DKilkenny committed:
Broyden

Commit:
Auto IVC bug fixes - set 1

tadkollar committed:
Merge pull request #95 from OpenMDAO/auto_ivc

Auto ivc

tadkollar committed:
Merge pull request #96 from tadkollar/master

sync

Kenneth-T-Moore committed:
Merge pull request #9 from DKilkenny/auto_ivc_docs

More Auto ivc docs

DKilkenny committed:
missed one test on nonlinear_block_gs

DKilkenny committed:
LinearBlockGS and LinearBlockJac

tadkollar committed:
Fixed recorder test to not look for _auto_ivc

DKilkenny committed:
PETScKrylov

DKilkenny committed:
ScipyKrylov

Commit:
review

DKilkenny committed:
LinearUserDefined

DKilkenny committed:
ArmijoGoldsteinLS

DKilkenny committed:
BoundsEnforceLS

DKilkenny committed:
Merge conflicts

DKilkenny committed:
KrigingSurrogate

DKilkenny committed:
ResponseSurface

swryan committed:
Merge pull request #1499 from Kenneth-T-Moore/auto_ivc2

Auto ivc2 bug fixes - set 1.

DKilkenny committed:
Backtracking fixes

DKilkenny committed:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc_docs

swryan committed:
Merge pull request #1497 from tadkollar/n2autoivc

N2 diagram inputs with auto-IVC'd output sources are red

DKilkenny committed:
Added default values

DKilkenny committed:
Added default values part 2

naylor-b committed:
iprof fix

DKilkenny committed:
working

swryan committed:
Merge pull request #1501 from naylor-b/ken_auto_ivc

small iprof fix for auto_ivc branch

swryan committed:
more error checking & testing

tadkollar committed:
Merge pull request #97 from OpenMDAO/auto_ivc

Auto ivc

DKilkenny committed:
working

DKilkenny committed:
Fixed shape issue

swryan committed:
better error handling/messages

swryan committed:
Merge pull request #1500 from DKilkenny/auto_ivc_docs

Auto IVC Doc Changes: Solvers and Surrogates

swryan committed:
update docs

swryan committed:
Merge branch 'auto_ivc' into work

swryan committed:
clean up feature test

swryan committed:
use get_val/set_val in feature tests

DKilkenny committed:
one test breaking

swryan committed:
update test_feature_simple tests in test_problem for auto_ivc

DKilkenny committed:
Added error and test

DKilkenny committed:
Removed else

tadkollar committed:
Merge pull request #100 from OpenMDAO/auto_ivc

Auto ivc

DKilkenny committed:
Merge in auto_ivc

robfalck committed:
Updated recording/reading documentation examples to use AutoIVC.

Commit:
Raise errors for disallowed auto-ivc configs.

swryan committed:
add_product/add_magnitude work properly when called in configure()

Commit:
cleanup

DKilkenny committed:
working on test_group.py

Commit:
converting tests

Commit:
Review Bret

swryan committed:
add tests for dot/cross/mvp, calling add_product in configure()

Commit:
Review Steve

tadkollar committed:
N2 auto_ivc doc migration

DKilkenny committed:
removed for loop

tadkollar committed:
Fixed sellar component from test suite for auto_ivc

DKilkenny committed:
pep

Commit:
Driver docs converted

naylor-b committed:
updated docs/tests to remote IndepVarComps.  Fixed bug (missing auto_ivc mapping in compute_jacvec_product

swryan committed:
Merge pull request #1518 from swryan/docs

update test_feature_simple tests in test_problem for auto_ivc

naylor-b committed:
merged out

Commit:
fixing discrete

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc

Commit:
Added asserts to some GA doc tests.

DKilkenny committed:
test_distrib and test_group so far

DKilkenny committed:
cleaned up test

Commit:
working on it

Commit:
discrete desvars work with auto_ivc

Commit:
undo

Commit:
discrete desvars work with auto_ivc

swryan committed:
Merge pull request #1520 from DKilkenny/src_mismatch

Raise exception when src_indices shape does not match target input shape

swryan committed:
Merge pull request #1522 from Kenneth-T-Moore/auto_ivc2

Raise exceptions for disallowed auto-ivc configurations..

swryan committed:
Merge pull request #1525 from Kenneth-T-Moore/auto_ivc

Removed indepvarcomps from Driver docs.

swryan committed:
Merge pull request #1526 from naylor-b/auto_ivc_docs

updated docs/tests to remote IndepVarComps.

naylor-b committed:
fix for problem get_val with indices before setup

naylor-b committed:
Merge remote-tracking branch 'origin/auto_ivc' into auto_ivc_docs

Commit:
Advanced features -> auto_ivc

Commit:
Advanced features -> auto_ivc

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc2

robfalck committed:
changed placement of set_input_defaults in test_sqlite_reader

swryan committed:
Merge pull request #1529 from naylor-b/auto_ivc_docs

fix for problem set_val with indices before setup

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc2

robfalck committed:
fix discrepancy between inputs and outputs

DKilkenny committed:
Merge in latest

tadkollar committed:
Cleanups for testing

naylor-b committed:
updated examples for auto_ivc

naylor-b committed:
Merge remote-tracking branch 'origin/auto_ivc' into auto_ivc_docs

DKilkenny committed:
test distrib, parallel_groups, group so far

tadkollar committed:
Switched from testing _auto_ivc component to sub component

robfalck committed:
updated tests for the debugging feature docs

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc

swryan committed:
Merge pull request #1524 from tadkollar/n2docs1510

Removed IndepVarComp references, added promotes and set_input_defaults where necessary

naylor-b committed:
removed commented code

robfalck committed:
fixed typo and unnecessary comments

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc

swryan committed:
update feature tests/docs for Working with Components

swryan committed:
Merge branch 'auto_ivc' into docs

robfalck committed:
input/output fix

Commit:
auto_ivc conversion, advanced feature docs

swryan committed:
Merge pull request #1528 from Kenneth-T-Moore/auto_ivc3

Fix so that discrete variables work as driver des_vars.

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into auto_ivc2

swryan committed:
Merge branch 'auto_ivc' into work

DKilkenny committed:
finished changes

DKilkenny committed:
merge latest

swryan committed:
Merge pull request #1530 from Kenneth-T-Moore/auto_ivc

Advanced features -> auto_ivc

DKilkenny committed:
Fixed test

DKilkenny committed:
Restored indicies test

swryan committed:
Merge pull request #1492 from johnjasa/approx_flag

Added flag for `under_approx` and a new counter `iter_count_without_approx`

swryan committed:
Merge pull request #1503 from swryan/work

Allow the linear algebra components to each perform multiple calculations.

swryan committed:
Merge pull request #1531 from naylor-b/auto_ivc_docs

Updated examples for auto_ivc

swryan committed:
Merge pull request #1533 from swryan/docs

Doc Migration AutoIvc -> Core Features - Defining Components

swryan committed:
Merge pull request #1535 from DKilkenny/auto_ivc_docs

Doc Migration AutoIvc -> Core Features

swryan committed:
fix doc

hschilling committed:
Remove IVCs, use set_val, and get_val

swryan committed:
pass kwargs in IndepVarComp init

hschilling committed:
merged out and fixed conflicts

robfalck committed:
Merge branch 'auto_ivc' of https://github.com/OpenMDAO/OpenMDAO into case_recording_doc_update

robfalck committed:
merged out

swryan committed:
add options to indepvarcomp

swryan committed:
Merge pull request #1532 from robfalck/debugging_feature_update

Debugging feature examples update

hschilling committed:
Double checked for omissions, commented out items, and more

Commit:
test mod

Commit:
Discrete auto_ivcs should obey list_autoivcs arg in system.list_inputs

Commit:
Discrete auto_ivcs should obey list_autoivcs arg in system.list_inputs

hschilling committed:
Removed unneeded connect

Commit:
Merge branch 'auto_ivc' of github.com:OpenMDAO/OpenMDAO into rob_discrete_autoivc_bug

hschilling committed:
Removed unneeded set_input_defaults

hschilling committed:
replaced set_input_defaults with set_val's

swryan committed:
Merge pull request #1521 from robfalck/case_recording_doc_update

Updated recording/reading documentation examples to use AutoIVC.

hschilling committed:
replaced set_input_defaults with set_val's

swryan committed:
Merge pull request #1536 from swryan/work

support options on IndepVarComp

swryan committed:
Merge pull request #1537 from hschilling/I1508-building-blocks-docs

I1508 did more updating of tests in Building Blocks Components/Drivers/Solvers/Surrogates

swryan committed:
Merge branch 'master' into work

swryan committed:
Merge pull request #1539 from swryan/work

Merge Auto IVC Branch into Master

DKilkenny committed:
Added val arg to add_balance and doc

payoto committed:
Document prerequisite to Problem.record method

Document in the `Problem.record` doc-string that it must be called
after `Problem.final_setup`. Otherwise the following error gets thrown:

    AttributeError: 'Problem' object has no attribute
    '_filtered_vars_to_record'

swryan committed:
add error message and test

swryan committed:
Merge pull request #1542 from DKilkenny/balance

Added val to BalanceComp Constructor

swryan committed:
update error message

swryan committed:
Merge pull request #1545 from swryan/record

Problem.record() provides a useful error message if called before final_setup.

Commit:
Merge branch 'master' of github.com:OpenMDAO/OpenMDAO into auto_ivc2

Commit:
Restoring wing project template.
